### ISSUE TITLE
Script hangs when trying to make a Net::HTTP request

### ISSUE THREAD

**Issue by saluzafa** [dsh0416/evt#6]
Hello,

I was playing with your gem, but I'm facing a little problem. I'm trying to make a HTTP request but it hangs after 5 seconds (read timeout) with this strange error:

```
/Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/protocol.rb:219:in `wait_readable': no implicit conversion of true into Integer (TypeError)
```

```ruby
require 'evt'
require 'net/http'

rd, wr = IO.pipe
scheduler = Evt::Scheduler.new

Fiber.set_scheduler scheduler

Fiber.schedule do
  url = URI.parse('https://jsonip.com')
  request = Net::HTTP::Get.new('/')
  response = Net::HTTP.start(url.host, url.port) do |http|
    http.read_timeout = 5
    http.request(request)
  end
  message = rd.read(20)
  puts message
  rd.close
end

Fiber.schedule do
  wr.write("Hello World")
  wr.close
end
```

Full trace:
```
/Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/protocol.rb:219:in `wait_readable': no implicit conversion of true into Integer (TypeError)
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/protocol.rb:219:in `rbuf_fill'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/protocol.rb:193:in `readuntil'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/protocol.rb:203:in `readline'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http/response.rb:42:in `read_status_line'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http/response.rb:31:in `read_new'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:1557:in `block in transport_request'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:1548:in `catch'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:1548:in `transport_request'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:1521:in `request'
	from test.rb:14:in `block (2 levels) in <main>'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:960:in `start'
	from /Users/slz/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http.rb:621:in `start'
	from test.rb:12:in `block in <main>'
```

Is this a bug or am I using it incorrectly?

Cheers :)

**Comment by dsh0416** [comment]
Nice catch. Investigating the bug...

**Comment by saluzafa** [comment]
Hi again :)

It seems that the problem comes from here (in `evt/bundled.rb`):
```ruby
  # Wait for the given file descriptor to match the specified events within
  # the specified timeout.
  # @parameter event [Integer] A bit mask of `IO::READABLE`,
  #   `IO::WRITABLE` and `IO::PRIORITY`.
  # @parameter timeout [Numeric] The amount of time to wait for the event in seconds.
  # @returns [Integer] The subset of events that are ready.
  def io_wait(io, events, duration)
    # TODO: IO::PRIORITY
    @readable[io] = Fiber.current unless (events & IO::READABLE).zero?
    @writable[io] = Fiber.current unless (events & IO::WRITABLE).zero?
    self.register(io, events)
    Fiber.yield
    self.deregister(io)
    true
  end
```

`#io_wait` should returns an Integer but that's not actually the case and makes `IO#wait_readable` crash.

I don't know what should be returned instead of that `Boolean`.

Cheers :)

**Comment by dsh0416** [comment]
@saluzafa Every IO halt comes from here. Would you please tell me your OS version, so that I could find out how the C API is called.

**Comment by dsh0416** [comment]
See 8b65415e5976ddd1ce2f4d0df0f363df7572c252, this `true` should not be the cause of this problem. I don't even believe that this return value is getting read by any method since the registration process is very simple in current Fiber Scheduler API. The problem should be caused somewhere deeply hidden in the MRI runtime.

**Comment by dsh0416** [comment]
Fixed in 7b1beba3e02461c627f5c85969658cc7852f15d6. Linux should be working. Still fixing mac kqueue.

**Comment by zt2** [comment]
Still got a problem with kqueue on mac:

```ruby
require 'evt'
require 'net/http'

scheduler = Evt::Scheduler.new

Fiber.set_scheduler scheduler

100.times do 
  Fiber.schedule do
    p Net::HTTP.get('httpbin.org', '/ip')
  end
end
```

result:

```
~/.rvm/gems/ruby-3.0.0/gems/evt-0.4.0/lib/evt/backends/kqueue.rb:24:in `kqueue_wait': execution expired (Net::OpenTimeout)
	from ~/.rvm/gems/ruby-3.0.0/gems/evt-0.4.0/lib/evt/backends/kqueue.rb:24:in `wait'
	from ~/.rvm/gems/ruby-3.0.0/gems/evt-0.4.0/lib/evt/backends/bundled.rb:39:in `run'
	from ~/.rvm/gems/ruby-3.0.0/gems/evt-0.4.0/lib/evt/backends/bundled.rb:149:in `close'
```
